rest/client.go

```diff
@@ -113,176 +122,178 @@ func (c *client) DeleteNode(ctx context.Context, id uuid.UUID) (err error) {
 	return
 }
 
-func (c *client) DuplicateNode(ctx context.Context, source uuid.UUID, destination uuid.UUID) (uuid.UUID, error) {
-	request := rest.Post("nodes/{node}/duplicate{?dstParentNodeId}").
+func (c *client) DuplicateNode(ctx context.Context, source uuid.UUID, destination uuid.UUID, suffix string) (models.DuplicateNodeResponse, error) {
+	request := rest.Post("nodes/{node}/duplicate{?dstParentNodeId,label_suffix}").
 		Assign("node", source).
 		Assign("dstParentNodeId", destination).
+		Assign("label_suffix", suffix).
 		SetHeader("Accept", "application/json")
 
-	var response models.WebmodelsNodeID
+	var response models.DuplicateNodeResponse
 	if err := c.DoAndUnmarshal(ctx, request, &response); err != nil {
-		return uuid.EmptyUUID, err
+		return models.DuplicateNodeResponse{}, err
 	}
 
-	return response.NodeID, nil
+	return response, nil
 }
 
-func (c *client) GetAncestors(ctx context.Context, id uuid.UUID, height int, nodeTypes ...string) ([]models.Node, error) {
+func (c *client) GetAncestors(ctx context.Context, id uuid.UUID, height int, nodeTypes ...string) (models.GetAncestorsResponse, error) {
 	request := rest.Get("nodes/{node}/ancestors{?height,type*}").
 		Assign("node", id).
 		Assign("height", height).
 		Assign("type", nodeTypes).
 		SetHeader("Accept", "application/json")
 
-	var response models.WebmodelsNodes
+	var response models.GetAncestorsResponse
 	if err := c.DoAndUnmarshal(ctx, request, &response); err != nil {
-		return nil, err
+		return models.GetAncestorsResponse{}, err
 	}
 
-	return response.Nodes, nil
+	return response, nil
 }
 
-func (c *client) GetCompany(ctx context.Context, id uuid.UUID) (models.Node, error) {
+func (c *client) GetCompany(ctx context.Context, id uuid.UUID) (models.GetCompanyResponse, error) {
 	request := rest.Get("nodes/{node}/company").
 		Assign("node", id).
 		SetHeader("Accept", "application/json")
 
-	var response models.WebmodelsNode
+	var response models.GetCompanyResponse
 	if err := c.DoAndUnmarshal(ctx, request, &response); err != nil {
-		return models.Node{}, err
+		return models.GetCompanyResponse{}, err
 	}
 
-	return response.Node, nil
+	return response, nil
 }
 
-func (c *client) GetSubtree(ctx context.Context, id uuid.UUID, filter TreeFilter) ([]models.Node, error) {
-	request := rest.Get("nodes/{node}/subtree{?depth,limit,offset,metadata_key,metadata_value,type*}").
+func (c *client) GetSubtree(ctx context.Context, id uuid.UUID, filter TreeFilter, continuationToken string) (models.GetSubtreeResponse, error) {
```

Comment:
Should this client handle pagination for the user? Since v2 enforces pagination I feel this can be nice, to avoid this code existing on each call site.

I was thinking more like the GetSubtree always returns all results (by looping over the endpoint and returning until the next URL is empty)

This should also apply to GetAncestors and the other endpoints that enforce pagination in v2.

Reply:
Get Ancestors doesn't have pagination does it. Created new funs for getsubtree and get origins.

Reply:
I feel we should just expose GetWholeSubtree as GetSubtree? Is there a use case for just getting one page?

Reply:
Well I figure pagination is a feature, why wouldn't we want the users to able to use all features?

Reply:
Pagination is something we force them to use, to be able to avoid big database queries. The only thing I can come up with is if they only want the first X things. But that can be modeled in the same method with a check against the "limit" parameter.

Because with your proposal as it is now the consumer needs to parse the `next` URL from the response to retrieve the continuation token?

Reply:
hrm ok so we don't return the token in any usable format. Iguess that makes it kind of mott to have the function withe token as input...